src/main/java/com/synopsys/integration/alert/component/settings/SettingsDescriptorActionApi.java

```diff
@@ -132,17 +132,22 @@ private void saveEncryptionProperties(final FieldModel fieldModel) {
     }
 
     private void addSAMLMetadata(final FieldModel fieldModel) {
-        // TODO enable SAML support
-        /*final Boolean samlEnabled = fieldModel.getFieldValueModel(SettingsDescriptor.KEY_SAML_ENABLED)
+        final Boolean samlEnabled = fieldModel.getFieldValueModel(SettingsDescriptor.KEY_SAML_ENABLED)
                                         .map(fieldValueModel -> fieldValueModel.getValue()
                                                                     .map(BooleanUtils::toBoolean)
                                                                     .orElse(false)
                                         ).orElse(false);
         final Optional<FieldValueModel> metadataURLFieldValueOptional = fieldModel.getFieldValueModel(SettingsDescriptor.KEY_SAML_METADATA_URL);
-        if (metadataURLFieldValueOptional.isPresent()) {
+        final Optional<FieldValueModel> metadataEntityFieldValueOptional = fieldModel.getFieldValueModel(SettingsDescriptor.KEY_SAML_ENTITY_ID);
+        final Optional<FieldValueModel> metadataBaseURLFieldValueOptional = fieldModel.getFieldValueModel(SettingsDescriptor.KEY_SAML_ENTITY_BASE_URL);
+        if (metadataURLFieldValueOptional.isPresent() && metadataEntityFieldValueOptional.isPresent() && metadataBaseURLFieldValueOptional.isPresent()) {
             final FieldValueModel metadataURLFieldValue = metadataURLFieldValueOptional.get();
+            final FieldValueModel metadataEntityFieldValue = metadataEntityFieldValueOptional.get();
+            final FieldValueModel metadataBaseUrValueModel = metadataBaseURLFieldValueOptional.get();
             final String metadataURL = metadataURLFieldValue.getValue().orElse("");
-            samlManager.updateSAMLConfiguration(samlEnabled, metadataURL);
-        }*/
+            final String entityId = metadataEntityFieldValue.getValue().orElse("");
+            final String baseUrl = metadataBaseUrValueModel.getValue().orElse("");
```

Comment:
Can we combine these three calls into the initial gets above?

Reply:
I will be writing more tests around the classes in a separate PR I will clean up this code then